src/profile-logic/tracks.js

```diff
@@ -744,180 +723,169 @@ export function getLocalTrackName(
   }
 }
 
-/**
- * Determine if a thread is idle, so that it can be hidden. It is really annoying for an
- * end user to load a profile full of empty and idle threads. This function uses
- * various rules to determine if a thread is idle.
- */
-function _isThreadIdle(
-  profile: Profile,
-  threadIndex: ThreadIndex,
-  idleThreadsByCPU: Set<ThreadIndex>
-): boolean {
-  const thread = profile.threads[threadIndex];
-  if (
-    // Don't hide the Renderer thread. This is because Renderer thread is pretty
-    // useful for understanding the painting with WebRender and it's an important
-    // thread for the users.
-    thread.name === 'Renderer' ||
-    // Don't hide the main thread of the parent process.
-    (thread.name === 'GeckoMain' && thread.processType === 'default') ||
-    // Don't hide the GPU thread on Windows.
-    (thread.name === 'GeckoMain' && thread.processType === 'gpu')
-  ) {
-    return false;
-  }
+// Consider threads whose sample score is 20% or less of the maximum sample score to be idle.
+const IDLE_THRESHOLD_FRACTION = 0.2;
 
-  if (thread.samples.length === 0) {
-    // This is a profile without any sample (taken with no periodic sampling mode)
-    // and we can't take a look at the samples to decide whether that thread is
-    // active or not. So we are checking if we have a paint marker instead.
-    return isThreadWithNoPaint(thread);
+// Return a non-empty set of threads that should be shown by default.
+export function computeDefaultVisibleThreads(
+  profile: Profile
+): Set<ThreadIndex> {
+  const threads = profile.threads;
+  if (threads.length === 0) {
+    throw new Error('No threads');
```

Comment:
Do we need to throw here? It's unlikely that we will have a profile without any threads, but we can maybe just return an empty thread instead?

Reply:
You mean "empty set"? The function promises that the set is non-empty. And I think we can never hit that exception because we have another check earlier in the pipeline that rejects profiles without threads.

Reply:
Ah sorry, yeah I wanted to write "empty set" and not thread. I agree that we can never hit this.